kotlin-reference-server/src/main/kotlin/org/stellar/reference/callbacks/rate/RateRoute.kt

```diff
@@ -0,0 +1,53 @@
+package org.stellar.reference.callbacks.rate
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.stellar.anchor.api.callback.GetRateRequest
+import org.stellar.anchor.util.GsonUtils
+import org.stellar.reference.callbacks.BadRequestException
+import org.stellar.reference.callbacks.NotFoundException
+import org.stellar.reference.callbacks.UnprocessableEntityException
+import org.stellar.reference.log
+
+/**
+ * Defines the routes related to the rate callback API. See
+ * [Rate Callbacks](https://developers.stellar.org/api/anchor-platform/callbacks/rate/).
+ *
+ * @param rateService the [RateService] to use to process the requests.
+ */
+fun Route.rate(rateService: RateService) {
+  authenticate("integration-auth") {
+    get("/rate") {
+      val request =
+        GetRateRequest.builder()
+          .type(call.parameters["type"]?.let { GetRateRequest.Type.valueOf(it.uppercase()) })
+          .sellAsset(call.parameters["sell_asset"])
+          .sellAmount(call.parameters["sell_amount"])
+          .sellDeliveryMethod(call.parameters["sell_delivery_method"])
+          .buyAsset(call.parameters["buy_asset"])
+          .buyAmount(call.parameters["buy_amount"])
+          .buyDeliveryMethod(call.parameters["buy_delivery_method"])
+          .countryCode(call.parameters["country_code"])
+          .expireAfter(call.parameters["expire_after"])
+          .clientId(call.parameters["client_id"])
+          .id(call.parameters["id"])
```

Comment:
❓ `id` is not a parameter to our `GET /rate` callback, is this used by our integration tests?

Reply:
The `id` is used to store quote at the reference server. I think we need more time to discuss whether to store the quote at AP or at the business server.